wkconnect/backends/neuroglancer/meshes.py

```diff
@@ -0,0 +1,173 @@
+from dataclasses import dataclass, replace
+from io import BytesIO
+from typing import Any, List, Tuple
+
+import DracoPy
+import numpy as np
+from stl import Mesh
+
+from ...utils.types import Vec3Df
+from .sharding import ShardingInfo
+
+
+@dataclass(frozen=True)
+class MeshfileFragment:
+    position: Vec3Df
+    byte_offset: int
+    byte_size: int
+
+
+@dataclass(frozen=True)
+class MeshfileLod:
+    scale: int
+    vertex_offset: Vec3Df
+    chunk_shape: Vec3Df
+    fragments: List[MeshfileFragment]
+
+
+@dataclass(frozen=True)
+class Meshfile:
+    chunk_shape: Vec3Df
+    grid_origin: Vec3Df
+    lods: List[MeshfileLod]
+
+    def fragment_offset_and_shape(
+        self, lod: int, fragment: int
+    ) -> Tuple[Vec3Df, Vec3Df, Vec3Df]:
+        return (
+            self.grid_origin
+            + self.lods[lod].fragments[fragment].position
+            * self.chunk_shape
+            * self.lods[lod].scale,
+            self.grid_origin
+            + self.lods[lod].fragments[fragment].position
+            * self.chunk_shape
+            * self.lods[lod].scale
+            + self.chunk_shape * self.lods[lod].scale,
+            self.chunk_shape * self.lods[lod].scale,
+        )
+
+    def decode_data(self, fragment: MeshfileFragment, buf: bytes) -> bytes:
+        mesh_obj = DracoPy.decode_buffer_to_mesh(buf)
+        vertices = np.array(mesh_obj.points, dtype="f4").reshape((-1, 3))
+        faces = np.array(mesh_obj.faces, dtype="<u4").reshape((-1, 3))
+        stl_mesh = Mesh(np.zeros(faces.shape[0], dtype=Mesh.dtype))
+        for i, f in enumerate(faces):
+            for j in range(3):
+                stl_mesh.vectors[i][j] = fragment.position.as_np() + vertices[f[j], :]
+        out_buf = BytesIO()
+        stl_mesh.save("mesh.stl", out_buf, update_normals=True)
+        return out_buf.getvalue()
+
+
+@dataclass(frozen=True)
+class MeshInfo:
+    sharding: ShardingInfo
+    _transform: Tuple[float, ...]
+    vertex_quantization_bits: int
+    lod_scale_multiplier: int = 1
+
+    @staticmethod
+    def parse(info_json: Any) -> "MeshInfo":
+        assert info_json["@type"] == "neuroglancer_multilod_draco"
+        return MeshInfo(
+            sharding=ShardingInfo.parse(info_json["sharding"]),
+            _transform=tuple(info_json["transform"]),
+            vertex_quantization_bits=info_json["vertex_quantization_bits"],
+            lod_scale_multiplier=info_json["lod_scale_multiplier"],
+        )
+
+    @property
+    def transform(self) -> np.ndarray:
+        return np.array(self._transform, dtype="f4").reshape((3, 4))
+
+    def parse_meshfile(self, buf: bytes, byte_offset: int) -> Meshfile:
```

Comment:
I guess this method is based on the documentation of the format? Is it this? https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md ? Could we link it here in a comment?